routes/auth.js

```diff
@@ -0,0 +1,191 @@
+// Import dependencies
+const smtpTransport = require('nodemailer-smtp-transport');
+const moment = require('moment');
+const config = require('../config/settings');
+const _      = require('lodash');
+const jwt    = require('jsonwebtoken');
+const authenticationHelpers = require('./authenticationHelpers');
+
+// create reset password token
+function createToken(user) {
+	return jwt.sign(_.omit(user, 'password'), config.secretKey, { expiresIn: 60*60*5 });
+}
+
+module.exports = function(router, passport, async, nodemailer, crypto, models){
+
+	const localUser = models.user_account;
+
+	router.post('/login', function(req, res, next) {
+		passport.authenticate('local-login', function(err, user, info) {
+			if (err) { 
+				return next(err); 
+			}
+			if (!user) {
+				return res.status(400).json(info); 
+			}
+			req.logIn(user, function(err) {
+				if (err) { 
+					return next(err); 
+				}
+				return res.status(200).json({"user": user,"token":createToken(user)});
+			});
+		})(req, res, next);
+	});
+	
+	router.get('/google', passport.authenticate('google', {scope: ['profile', 'email']}));
+
+	router.get('/google/callback', 
+		passport.authenticate('google', { successRedirect: '/home',
+			failureRedirect: '/' }));
+
+	router.get('/logout', authenticationHelpers.isAuthOrRedirect, function(req, res, next) {
+		req.logout();
+		res.json({'loggedOut': req.isAuthenticated()});
+	});
+
+	router.get('/authenticated', authenticationHelpers.isAuth, function(req, res, next) {
+		res.json({'authenticated': true,'token':createToken(req.user)});
+	});	                                      	
+	
+	// Forgot password and generate token
+	router.post('/forgot', function(req, res, next){
+		async.waterfall([
+			function(done) {
+				crypto.randomBytes(20, function(err, buf) {
+					const token = buf.toString('hex');
+					done(err, token);
+				});
+			},
+			function(token, done) {
+				const user = {email:req.body.email};
+				localUser.find({where: {
+					email: req.body.email,
+					provider: 'local'
+				}}, {raw: true})
+				.then(data=> {
+					if(!data){
+						return res.status(400).json({'error':'This account does not exist or you cannot change the password for this account'});			
+					}				
+					let date = moment(moment.now() + 3600000).format('YYYY-MM-DD HH:mm:ss');
```

Comment:
3600000 is the milliseconds equivalent for one hour.

I think I am using date again somewhere, will check it out and make the necessary change.